### PR TITLE
:bug: Fix position of context menu off the screen

### PR DESCRIPTION
**Pull request by hirunatan**

### CODE REVIEW

File: frontend/src/uxbox/main/ui/workspace/context_menu.cljs [hunk 1]
```diff
@@ -124,12 +124,29 @@
...
+      #(let [dropdown (mf/ref-val dropdown-ref)]
+         (when dropdown
+           (let [bounding-rect (dom/get-bounding-rect dropdown)
+                 window-size (dom/get-window-size)
+                 delta-x (max (- (:right bounding-rect) (:width window-size)) 0)
+                 delta-y (max (- (:bottom bounding-rect) (:height window-size)) 0)
+                 new-style (str "top: " (- top delta-y) "px; "
+                                "left: " (- left delta-x) "px;")]
+             (when (or (> delta-x 0) (> delta-y 0))
+               (.setAttribute dropdown "style" new-style))))))
```
Comment (niwinz) [inline frontend/src/uxbox/main/ui/workspace/context_menu.cljs]: I think we need hint dropdown as `^js` for avoid unexpected exception on production builds.

File: frontend/src/uxbox/util/dom.cljs [hunk 2]
```diff
@@ -162,6 +162,19 @@
   {:width (.-clientWidth ^js node)
    :height (.-clientHeight ^js node)})
 
+(defn get-bounding-rect
+  [node]
+  (let [rect (.getBoundingClientRect ^js node)]
```
Comment (niwinz) [inline frontend/src/uxbox/util/dom.cljs:167]: the same here, `^js` meta tag is needed  on all interop calls.

File: frontend/src/uxbox/main/ui/workspace/context_menu.cljs (hunk 1 above)
Comment (hirunatan) [reply frontend/src/uxbox/main/ui/workspace/context_menu.cljs]: done

File: frontend/src/uxbox/util/dom.cljs (hunk 2 above)
Comment (hirunatan) [reply frontend/src/uxbox/util/dom.cljs:167]: Changed. Not sure if well done. Other functions in this module do not always have the tag.

Comment (niwinz) [reply frontend/src/uxbox/util/dom.cljs:167]: Many interop calls are already covered by builtin externs. but is a good practice tagging all the interpor calls for avoid future headaches with the production builds.

**Pull request merged**